x/genutil/client/cli/init.go

```diff
@@ -66,6 +66,8 @@ func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx.HomeDir, _ = cmd.Flags().GetString(flags.FlagHome)
```

Comment:
@alexanderbez is this the correct place to take the home result?

Reply:
I really home we aren't going to have to read every flag in every command. I think at a minimum we should have helpers to read those flags.

Reply:
Yeah I agree, we need to gather which flags are used by every command and this will help.

I am worried that this can become the AddQueryFlagsToCmd or AddTxFlagsToCmd that I have the sensation that not all flags are used by every command that benefits from this functions. (maybe is my overthinking :D)

Reply:
I'm not sure what's going on here, but you don't need to do this. `clientCtx.HomeDir` is already set due to the call to `GetClientContextFromCmd`.

Look at the flow, `GetClientContextFromCmd` reads a context, that context will already have a homedir set (see `SetCmdClientContextHandler`).

Reply:
> I really home we aren't going to have to read every flag in every command. I think at a minimum we should have helpers to read those flags.

You don't -- that's why we have `GetClientContextFromCmd`, `ReadQueryCommandFlags`, and `ReadTxCommandFlags`.

Reply:
Ok gonna check, because for some reason home was not populated by default.

Reply:
Is this in a test? If so, that is because it has to be set on the provided `client.Context`

Reply:
No no. Was in simd.